pkg/cmd/util/clientcmd/factory.go

```diff
@@ -45,12 +47,51 @@ func New(flags *pflag.FlagSet) *Factory {
 
 	// TODO: there should be two client configs, one for OpenShift, and one for Kubernetes
 	clientConfig := DefaultClientConfig(flags)
+	clientConfig = defaultingClientConfig{clientConfig}
 	f := NewFactory(clientConfig)
 	f.BindFlags(flags)
 
 	return f
 }
 
+// defaultingClientConfig detects whether the provided config is the default, and if
+// so returns an error that indicates the user should set up their config.
+type defaultingClientConfig struct {
+	nested kclientcmd.ClientConfig
+}
+
+// RawConfig calls the nested method
+func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
+	return c.nested.RawConfig()
+}
+
+// Namespace calls the nested method
+func (c defaultingClientConfig) Namespace() (string, bool, error) {
+	return c.nested.Namespace()
```

Comment:
add a TODO to figure out how to get namespace for in-cluster config... the upstream way isn't worth copying, imo, but we'll likely want to do something